uint8 state
float32 value
---
uint8 result
